src/gmt_io.c

```diff
@@ -824,17 +824,30 @@ GMT_LOCAL bool gmtio_ogr_header_parser (struct GMT_CTRL *GMT, char *record) {
 				break;
 
 			case 'J':	/* Dataset projection strings (one of 4 kinds) */
+				if (p[1] == '\0' || p[1] == ' ' || strchr ("egpw", p[1]) == NULL) {
+					GMT_Report (GMT->parent, GMT_MSG_ERROR, "Bad OGR/GMT: @J given unknown format (%c)\n", (int)p[1]);
+					GMT->current.io.ogr = GMT_OGR_FALSE;
+					return (false);					
+				}
+				Jstring = strdup (&p[2]);
 				switch (p[1]) {
-					case 'e': k = 0;	break;	/* EPSG code */
-					case 'g': k = 1;	break;	/* GMT proj code */
-					case 'p': k = 2;	break;	/* Proj.4 code */
-					case 'w': k = 3;	break;	/* OGR WKT representation */
-					default:
-						GMT_Report (GMT->parent, GMT_MSG_ERROR, "Bad OGR/GMT: @J given unknown format (%c)\n", (int)p[1]);
-						GMT->current.io.ogr = GMT_OGR_FALSE;
-						return (false);
+					case 'e': k = 0;	/* EPSG code */
+						if (strstr (Jstring, "4326"))	/* 4326 means we have lon/lat degree coordinates and thus should set -fig */
+							gmt_set_geographic (GMT, GMT_IN);
+						break;
+					case 'g': k = 1;	break;	/* GMT proj code. If given then data are projected */
+					case 'p': k = 2;	/* Proj.4 code */
+						if (strstr (Jstring, "+proj=longlat"))	/* +proj=longlat means we have lon/lat degree coordinates and thus should set -fig */
+							gmt_set_geographic (GMT, GMT_IN);
+						break;
+					case 'w': k = 3;	/* OGR WKT representation */
+						if (strstr (Jstring, "Degree"))	/* UNIT as Degree means we have lon/lat degree coordinates and thus should set -fig */
```

Comment:
Nope. It would be ``degree`` but I think many many WKTs have ``UNIT["degree",0.0174532925199433``.
AS I said, the only way I know out of this is to convert the WKT to PROJ4 and parse that.